lms/djangoapps/teams/views.py

```diff
@@ -0,0 +1,508 @@
+
+                * language: Optionally specifies which language the team is
+                  associated with.
+
+                * membership: A list of the users that are members of the team. See
+                  membership endpoint for more detail.
+
+            For all text fields, clients rendering the values should take care
+            to HTML escape them to avoid script injections, as the data is
+            stored exactly as specified. The intention is that plain text is
+            supported, not HTML.
+
+            If the user is not logged in or is not staff for the course, a 403 error
+            is returned.
+
+            If the specified team does not exist, a 404 error is returned.
+
+        **Response Values for PATCH**
+
+            Only staff can patch teams. If the specified team does not exist, a
+            404 error is returned. If the user is not staff, a 403 error is
+            returned.
+
+            If "application/merge-patch+json" is not the specified content type,
+            a 415 error is returned.
+
+            If the update could not be completed due to validation errors, this
+            method returns a 400 error with all error messages in the
+            "field_errors" field of the returned JSON.
+    """
+
+    class IsEnrolledOrIsStaff(permissions.BasePermission):
+        """Permission that checks to see if the user is enrolled in the course or is staff."""
+
+        def has_object_permission(self, request, view, obj):
+            """Returns true if the user is enrolled or is staff."""
+            return CourseEnrollment.is_enrolled(request.user, obj.course_id) or request.user.is_staff
+
+    authentication_classes = (SessionAuthentication, OAuth2Authentication)
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrReadOnly, IsEnrolledOrIsStaff,)
+    lookup_field = 'team_id'
+    serializer_class = CourseTeamSerializer
+    parser_classes = (MergePatchParser,)
+
+    def get_queryset(self):
+        """Returns the queryset used to access the given team."""
+        return CourseTeam.objects.all()
+
+
+class TopicListView(GenericAPIView):
+    """
+        **Use Cases**
+
+            Retrieve a list of topics associated with a single course.
+
+        **Example Requests**
+
+            GET /api/team/v0/topics/?course_id={course_id}
+
+        **Query Parameters for GET**
+
+            * course_id: Filters the result to topics belonging to the given
+              course (required).
+
+            * order_by: Orders the results. Currently only 'name' is supported,
+              and is also the default value.
+
+            * page_size: Number of results to return per page.
+
+            * page: Page number to retrieve.
+
+        **Response Values for GET**
+
+            If the course_id is not given or an unsupported value is passed for
+            order_by, returns a 400 error.
+
+            If the user is not logged in or not enrolled in the course, returns
```

Comment:
I think we will want "staff" == "is_staff + course staff" to be able to call this method.